cli/internal/plugins/git/versioner.go

```diff
@@ -0,0 +1,72 @@
+package git
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/gobuffalo/buffalo-cli/plugins"
+)
+
+type Versioner struct {
+	pluginsFn plugins.PluginFeeder
+}
+
+func (b *Versioner) WithPlugins(f plugins.PluginFeeder) {
+	b.pluginsFn = f
+}
+
+func (b *Versioner) ScopedPlugins() []plugins.Plugin {
+	if b.pluginsFn == nil {
+		return []plugins.Plugin{}
+	}
+
+	var scoped []plugins.Plugin
+	for _, p := range b.pluginsFn() {
+		switch p.(type) {
+		case VersionRunner:
+			scoped = append(scoped, p)
+		}
+	}
+
+	return scoped
+}
+
+func (b *Versioner) BuildVersion(ctx context.Context, root string) (string, error) {
+	if _, err := exec.LookPath("git"); err != nil {
+		return "", err
+	}
+
+	ioe := plugins.CtxIO(ctx)
+
+	cmd := exec.CommandContext(ctx, "git", "rev-parse", "--short", "HEAD")
+	cmd.Stdin = ioe.Stdin()
+	cmd.Stdout = ioe.Stdout()
+	cmd.Stderr = ioe.Stderr()
+
+	for _, p := range b.ScopedPlugins() {
+		if vr, ok := p.(VersionRunner); ok {
```

Comment:
Aha! The live command will still go the regular route. 🤔 Does that mean that the only implementations of this interface will be tests? I feel both cases, `CLI` and `string/mock` should fall under the same interface, otherwise, it feels like an if on steroids.